From 8a7e44ccf03fc1b10c43f21595f249a4a5061dfd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Mon, 2 Sep 2019 14:01:36 +0200
Subject: [PATCH 019/152] extensionSystem: Use logError to record extension
 errors with stack trace

Extensions might emit JS errors explicitly or implicitly, however GNOME
Shell doesn't present any stack trace for those making them quite hard
to debug.

Make this easier by logging errors with logError() whichs includes the
stack dump.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/700
---
 js/ui/extensionSystem.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/extensionSystem.js b/js/ui/extensionSystem.js
index fd7196972..a668907f7 100644
--- a/js/ui/extensionSystem.js
+++ b/js/ui/extensionSystem.js
@@ -194,7 +194,7 @@ var ExtensionManager = class {
             extension.errors = [];
         extension.errors.push(message);
 
-        log('Extension "%s" had error: %s'.format(uuid, message));
+        logError(error, `Extension ${uuid}`);
         this.emit('extension-state-changed', extension);
     }
 
-- 
2.23.0

